src/OrchardCore.Modules/OrchardCore.Users/Views/Items/ValidateUserTask.Fields.Edit.cshtml

```diff
@@ -0,0 +1,14 @@
+@model OrchardCore.Users.Workflows.ViewModels.ValidateUserTaskViewModel
+
+<fieldset class="form-group form-check" asp-validation-class-for="SetUser">
+    <input type="checkbox" asp-for="SetUser" class="form-check-input" />
+    <span asp-validation-for="SetUser"></span>
+    <label asp-for="SetUser" class="form-check-label">@T["Set the User property if the user is Authenticated"]</label>
+</fieldset>
+
+<fieldset class="form-group" asp-validation-class-for="Roles">
+    <label asp-for="Roles">@T["Roles"]</label>
+    @await Component.InvokeAsync("SelectRoles", new { selectedRoles = Model.Roles, htmlName = Html.NameFor(m => m.Roles), except = new[] { "Anonymous", "Authenticated" } })
```

Comment:
So maybe here, if we don't keep the except parameter

    @await Component.InvokeAsync("SelectRoles", new { selectedRoles = Model.Roles.Except(new[] { "Anonymous", "Authenticated" }), htmlName = Html.NameFor(m => m.Roles) })